manageiq-operator/pkg/helpers/miq-components/httpd.go

```diff
@@ -104,17 +112,23 @@ func NewHttpdConfigMap(cr *miqv1alpha1.ManageIQ) *corev1.ConfigMap {
 		"external-auth-openid-connect-remote-user-conf": string(externalAuthOpenidConnectRemoteUserConf),
 	}
 
-	return &corev1.ConfigMap{
+	ConfigMap := &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "httpd-configs",
 			Namespace: cr.ObjectMeta.Namespace,
 			Labels:    labels,
 		},
 		Data: data,
 	}
+
+	mutateF := func() error {
+		ConfigMap.Labels = labels
```

Comment:
I'm just going to comment here because it's a pretty simple case, but this applies generally.

I think that we should be setting everything that is important to us in this mutate function, so we should also be setting data.

Because of that we should deduplicate this logic and only set labels and data in the mutate function as it's called for both create and update. The only things we should need for the initial object are `Name` and `Namespace`.